### PR TITLE
refactor(bazel): Move all supported rules to the separate directory

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

Extract each rule to the separate file in dedicated directory

## Context

- Ref: #9667

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

File: lib/modules/manager/bazel/rules/docker.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  rule: depType,
+  name: depName,
+  tag: currentValue,
+  digest: currentDigest,
+  repository: packageName,
+  registry,
+}: Target): PackageDependency | null {
+  let dep: PackageDependency | null = null;
+
+  if (
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/docker.ts:17]: can we invert the condition, makes code flow easier

File: lib/modules/manager/bazel/rules/git.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export function gitDependency({
+  rule: depType,
+  name: depName,
+  tag: currentValue,
+  commit: currentDigest,
+  remote,
+}: Target): PackageDependency | null {
+  let dep: PackageDependency | null = null;
+
+  if (
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/git.ts:16]: same here.

File: lib/modules/manager/bazel/rules/go.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  rule: depType,
+  name: depName,
+  tag: currentValue,
+  commit: currentDigest,
+  importpath: packageName,
+  remote,
+}: Target): PackageDependency | null {
+  let dep: PackageDependency | null = null;
+
+  if (
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/go.ts:17]: same

File: lib/modules/manager/bazel/rules/http.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+export function httpDependency({
+  rule: depType,
+  name: depName,
+  url,
+  urls,
+  sha256,
+}: Target): PackageDependency | null {
+  let dep: PackageDependency | null = null;
+
+  if (
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/http.ts:62]: same

File: lib/modules/manager/bazel/rules/index.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+const supportedRules = Object.keys(dependencyExtractorRegistry);
+export const supportedRulesRegex = regEx(`^${supportedRules.join('|')}$`);
+
+function isTarget(x: Record<string, TargetAttribute>): x is Target {
+  return is.string(x.name) && is.string(x.rule);
+}
+
+export function coerceFragmentToTarget(fragment: Fragment): Target | null {
+  if (fragment.type === 'record') {
```
Comment (viceice) [inline lib/modules/manager/bazel/rules/index.ts:35]: same

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/bazel/rules/docker.ts (hunk 1 above)
Comment (zharinov) [reply lib/modules/manager/bazel/rules/docker.ts:17]: Let's do in separate PR

**Review by viceice — APPROVED** [review]

**Pull request merged**